### PR TITLE
Unify all the proof tests for rpc server that test invalid requests and make them table driven.

### PR DESCRIPTION
**Pull request by Martin2112**
Until it's been implemented. Update tests. Unify all the proof tests that test invalid requests and make them table driven.

### CODE REVIEW

File: server/log_rpc_server_test.go [hunk 1]
```diff
@@ -657,35 +698,22 @@ func TestGetLeavesByLeafValueHash(t *testing.T) {
...
-}
+		// Request should fail validation before any storage operations
+		mockStorage := storage.NewMockLogStorage(ctrl)
+		registry := testonly.NewRegistryWithLogProvider(mockStorageProviderFunc(mockStorage))
+		server := NewTrillianLogRPCServer(registry, fakeTimeSource)
 
-func TestGetProofByHashBadHash(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
+		_, err := server.GetInclusionProofByHash(context.Background(), test.hReq)
```
Comment (codingllama) [inline server/log_rpc_server_test.go]: Suggestion: Use if initialization?

(Same for others.)

Comment (Martin2112) [inline server/log_rpc_server_test.go]: Sadly it needs access to resp later and I was previously criticized for multiple else clauses in a similar situation.

**Comment by Martin2112** [comment]
This has been reworked to remove the API changes as I believe this feature can be implemented safely without it. I thought it was still worth having the test improvements.

File: server/log_rpc_server_test.go (hunk 1 above)
Comment (codingllama) [inline server/log_rpc_server_test.go]: I meant it basically as combining this and the if below, so it wouldn't need any more if/else clauses than we already have.

if _, err := server.GetInclusionProofByHash(context.Background(), test.hReq); err == nil || ... {
	t.Errorf("...")
}

(Just clarifying, please take it as a suggestion.)

**Comment by Martin2112** [comment]
OK the last round of changes might have moved the code you commented. I'll have another look.

**Comment by Martin2112** [comment]
That code is removed in #288 where I cleaned up all the failure mode proof tests.

**Pull request merged**